packages/shared/test-suites/RecordingTestMetadata.ts

```diff
@@ -660,54 +671,19 @@ export async function processGroupedTestCases(
           // we need to splice only the appropriate subset for each test.
           const annotationsByTest: Annotation[][] = annotations.reduce(
             (accumulated: Annotation[][], annotation: Annotation) => {
-              eventSwitch: switch (annotation.message.event) {
-                case "step:enqueue":
-                case "step:start": {
-                  if (currentTestHasEnded) {
-                    // TODO [SCS-1186]
-                    // Ignore steps that start outside of a test boundary;
-                    // These likely correspond to beforeAll or afterAll hooks which we filter for now
-                    return accumulated;
-                  }
-                  break;
-                }
-                case "test:start": {
-                  // Tests that were skipped won't have annotations.
-                  // Add empty annotations arrays for these.
-                  if (currentTestHasEnded) {
-                    while (currentTestRecordingIndex < partialTestRecordings.length - 1) {
-                      currentTestRecordingIndex++;
-                      currentTestRecording = partialTestRecordings[currentTestRecordingIndex];
-
-                      currentTestAnnotations = [];
-                      currentTestHasEnded = false;
-
-                      accumulated.push(currentTestAnnotations);
-
-                      if (currentTestRecording.result !== "skipped") {
-                        break eventSwitch;
-                      }
-                    }
-                  }
-
-                  const currentTest = partialTestRecordings[currentTestRecordingIndex];
-                  assert(
-                    currentTest?.id === annotation.message.testId,
-                    `Test id should match "test:start" annotation testId`
-                  );
-
-                  break;
-                }
-                case "test:end": {
-                  currentTestHasEnded = true;
-                  break;
-                }
-              }
-
-              // Ignore annotations that happen before the first test
-              // (These are probably beforeAll annotations, which we don't fully support yet)
-              if (currentTestAnnotations) {
-                currentTestAnnotations.push(annotation);
+              const { testId, attempt } = annotation.message;
+
+              assert(testId != null, "Annotation is missing `testId`. Plugin update required.");
+              assert(attempt != null, "Annotation is missing `attempt`. Plugin update required.");
+
+              // beforeAll/afterAll have -1 testId and can be ignored for now
+              if (testId !== -1) {
```

Comment:
Can we just _not_ store these events? Seems kind of yuck to have them included with a negative test id.

Reply:
I guess I get it if they aren't associated with a test, but `null` id seems more expected then.